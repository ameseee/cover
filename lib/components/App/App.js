import React from 'react';

const App = () => {
  return (
    <div className="app-container">
      I AM THE APP DIV!!!
    </div>
  );
};

export default App;
