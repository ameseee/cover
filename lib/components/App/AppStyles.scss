$red-color: #dc0800;
$turq-color: #039fbe;
$navy-color: #282a40;
$lt-grey-color: #f5f5f7;
$charcoal-color: #1b1b1b;
