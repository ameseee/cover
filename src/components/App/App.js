import React, { Component } from 'react';
import '../../index.css';
import { Link } from 'react-router-dom';
import firebase from '../../firebase';

class App extends Component {
  constructor() {
    super();
    this.signOut = this.signOut.bind(this);
  }

  signOut() {
    console.log('sign out props1:', this.props);
    firebase.auth().signOut().then( () => {
      //signout successful - this is prob where I should call props
    }).catch( error => {
      alert('We experienced an error:', error)
    });
    this.props.signOut(false);
    this.props.setCurrentUser('');
  }

  render() {
    return (
      <div className="app-container">

        <nav className="nav-bar">
          <h1 className="cover-title">
            <Link
              className="nav-bar-brand-link"
              to="/main"
            >
              Cover
            </Link>
          </h1>
          <ul className="nav-menu">
            <li className="nav-item-li">
              <Link
                className="nav-item"
                to="/contacts"
              >
              Manage Contacts
            </Link>
            </li>

            <li
              className="nav-item-li"
              onClick={this.signOut}>
              <Link
                className="nav-item"
                to="/auth"
              >
                Sign Out
              </Link>
            </li>

          </ul>
        </nav>
      </div>
    );
  };
}

export default App;
