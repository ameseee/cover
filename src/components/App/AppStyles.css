@import url(https://fonts.googleapis.com/css?family=Lato);
.hamburger {
  display: none;
  background-image: url("../../assets/hamburger.svg");
  background-size: cover;
  height: 25px;
  width: 25px;
  border: none;
  background-color: #293E5F;
  margin-right: 7.5px; }

.nav-bar-brand-link {
  text-decoration: none;
  color: #f5f5f7;
  font-family: Lato; }

.nav-bar-brand-link:hover {
  color: #47A7B4; }

.nav-bar {
  background-color: #293E5F;
  display: flex;
  align-items: center;
  justify-content: space-between; }

.cover-title {
  font-size: 170%;
  color: #fff;
  padding: 10px;
  font-family: Lato; }

.nav-menu {
  margin-right: 20px;
  display: flex;
  flex-direction: row;
  align-items: center; }

.nav-item {
  color: #fff;
  margin: 10px 15px;
  font-family: Lato;
  padding: 4px;
  border-bottom: 2px solid #293E5F;
  text-decoration: none; }

.nav-item-li {
  display: inline-block;
  list-style-type: none; }

.nav-item:hover {
  color: #E6E6E8;
  border-bottom: 2px solid #47A7B4;
  cursor: pointer; }

.sign-out {
  background-color: #47A7B4;
  color: #fff;
  border: none;
  border-radius: 3px;
  font-size: 90%;
  padding: 8px 12px;
  width: 90%;
  font-family: Lato;
  margin-left: 15px; }

.sign-out:hover {
  background-color: #2D9AA9; }

.sign-out-link {
  color: #fff;
  font-family: Lato;
  text-decoration: none; }

.exit {
  position: absolute;
  bottom: 10px;
  left: 10px;
  background-color: #de504b;
  padding: 10px;
  width: 10%;
  margin: 0 5% 0 auto;
  border: none;
  border-radius: 3px;
  color: #fff;
  font-size: 120%;
  font-family: Lato; }

.google-link,
.google-link:visited {
  color: #fff;
  text-decoration: none; }

@media (max-width: 670px) {
  .nav-menu {
    flex-direction: row-reverse;
    justify-content: space-between;
    width: 150px; }
  .nav-item-li {
    display: none; }
  .hamburger {
    display: block; }
  .hidden {
    display: none; }
  .sign-out {
    display: none; }
  .exit {
    width: 20%; } }
