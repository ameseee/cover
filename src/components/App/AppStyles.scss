@import url(https://fonts.googleapis.com/css?family=Lato);

$red-color: #dc0800;
$red-hover: #E54640;
$turq-color: #039fbe;
$turq-hover: #42B7CE;
$navy-color: #282a40;
$navy-hover: #434558;
$lt-grey-color: #f5f5f7;
$grey-color: #E6E6E8;
$charcoal-color: #1b1b1b;

.hamburger {
  display: none;
  color: #fff;
}

.hamburger {
  background-image: url('../../assets/hamburger.svg');
  background-size: cover;
  height: 25px;
  width: 25px;
  border: none;
  background-color: $navy-color;
}

.nav-bar-brand-link {
  text-decoration: none;
  color: $lt-grey-color;
  font-family: Lato;
}

.nav-bar-brand-link:hover {
  color: $turq-color;
}

.nav-bar {
  background-color: $navy-color;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.cover-title {
  font-size: 170%;
  color: #fff;
  padding: 10px;
  font-family: Lato;
}

.nav-menu {
  margin-right: 10px;
}

.nav-item {
  color: #fff;
  margin: 10px 15px;
  font-family: Lato;
  padding: 4px;
  border-bottom: 2px solid $navy-color;
  text-decoration: none;
}

.nav-item-li {
  display: inline-block;
  list-style-type: none;
}

.nav-item:hover {
  color: $grey-color;
  border-bottom: 2px solid $turq-color;
  cursor: pointer;
}


@media (max-width: 670px) {
  .nav-item-li {
    display: none;
  }

  .hamburger {
    display: block;
  }

}
