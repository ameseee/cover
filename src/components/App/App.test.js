import React from 'react';
import App from './App';
import { shallow } from 'enzyme';
import testSetup from '../../../__mock__/testSetup';

describe('<App />', () => {
  let wrapper;
  let noUserWrapper;
  let mockFn = jest.fn();

  beforeEach( () => {
    wrapper = shallow(<App currentUser="Amy"/>);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have a false default state of mobileNav', () => {
    expect(wrapper.state().mobileNav).toEqual(false);
  });

  it('should render a nav bar', () => {
    const nav = wrapper.find('nav');

    expect(nav.length).toEqual(1);
  });

  it('should render a brand title', () => {
    const title = wrapper.find('h1');

    expect(title.length).toEqual(1);
  });

  it('should render a nav bar list', () => {
    const navList = wrapper.find('ul');

    expect(navList.length).toEqual(1);
  });

  it('should render two options on nav bar if user is signed in', () => {
    const navItems = wrapper.find('li');

    expect(navItems.length).toEqual(2);
  });

  it('should render two options on nav bar if no user', () => {
    const noUserWrapper = shallow(<App currentUser=""/>);
    const navItems = noUserWrapper.find('li');

    expect(navItems.length).toEqual(2);
  });

  it('should render a sign out link if user is signed in', () => {
    const signOut = wrapper.find('.sign-out');

    expect(signOut.length).toEqual(1);
  });

  it('should change state on click of the hamburger', () => {
    const props = { history: [] };
    const wrapper = shallow(<App {...props}/>);
    const hamburger = wrapper.find('.hamburger');

    hamburger.simulate('click');
    expect(wrapper.state().mobileNav).toEqual(true);
  });

  it('should run a function on click ', () => {
    const props = { history: [] };
    const wrapper = shallow(<App {...props}/>);
    const hamburger = wrapper.find('.hamburger');

    wrapper.instance().showMobileNav = mockFn;
    hamburger.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should render an exit button at all times', () => {
    const exitBtn = wrapper.find('.exit');

    expect(exitBtn.length).toEqual(1);
  });

  it('should match snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});
