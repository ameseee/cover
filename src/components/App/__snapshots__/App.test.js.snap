// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    currentUser="Amy"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <nav
          className="nav-bar"
        >
          <h1
            className="cover-title"
          >
            <Link
              className="nav-bar-brand-link"
              replace={false}
              to="/main"
            >
              Cover
            </Link>
          </h1>
          <ul
            className="nav-menu"
          >
            <button
              className="hamburger"
              onClick={[Function]}
            />
            <li
              className="nav-item-li"
            >
              <Link
                className="nav-item"
                replace={false}
                to="/main"
              >
                Home
              </Link>
            </li>
            <li
              className="nav-item-li"
            >
              <Link
                className="nav-item"
                replace={false}
                to="/contacts"
              >
                Contacts
              </Link>
            </li>
            <button
              className="sign-out"
              onClick={[Function]}
            >
              <Link
                className="sign-out-link"
                replace={false}
                to="/"
              >
                Sign Out
              </Link>
            </button>
          </ul>
        </nav>,
        <button
          className="exit"
        >
          <a
            className="google-link"
            href="https://weather.com/"
          >
            EXIT
          </a>
        </button>,
      ],
      "className": "app-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="cover-title"
            >
              <Link
                className="nav-bar-brand-link"
                replace={false}
                to="/main"
              >
                Cover
              </Link>
            </h1>,
            <ul
              className="nav-menu"
            >
              <button
                className="hamburger"
                onClick={[Function]}
              />
              <li
                className="nav-item-li"
              >
                <Link
                  className="nav-item"
                  replace={false}
                  to="/main"
                >
                  Home
                </Link>
              </li>
              <li
                className="nav-item-li"
              >
                <Link
                  className="nav-item"
                  replace={false}
                  to="/contacts"
                >
                  Contacts
                </Link>
              </li>
              <button
                className="sign-out"
                onClick={[Function]}
              >
                <Link
                  className="sign-out-link"
                  replace={false}
                  to="/"
                >
                  Sign Out
                </Link>
              </button>
            </ul>,
          ],
          "className": "nav-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="nav-bar-brand-link"
                replace={false}
                to="/main"
              >
                Cover
              </Link>,
              "className": "cover-title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "Cover",
                "className": "nav-bar-brand-link",
                "replace": false,
                "to": "/main",
              },
              "ref": null,
              "rendered": "Cover",
              "type": [Function],
            },
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="hamburger"
                  onClick={[Function]}
                />,
                <li
                  className="nav-item-li"
                >
                  <Link
                    className="nav-item"
                    replace={false}
                    to="/main"
                  >
                    Home
                  </Link>
                </li>,
                <li
                  className="nav-item-li"
                >
                  <Link
                    className="nav-item"
                    replace={false}
                    to="/contacts"
                  >
                    Contacts
                  </Link>
                </li>,
                <button
                  className="sign-out"
                  onClick={[Function]}
                >
                  <Link
                    className="sign-out-link"
                    replace={false}
                    to="/"
                  >
                    Sign Out
                  </Link>
                </button>,
              ],
              "className": "nav-menu",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "hamburger",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-item"
                    replace={false}
                    to="/main"
                  >
                    Home
                  </Link>,
                  "className": "nav-item-li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Home",
                    "className": "nav-item",
                    "replace": false,
                    "to": "/main",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-item"
                    replace={false}
                    to="/contacts"
                  >
                    Contacts
                  </Link>,
                  "className": "nav-item-li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Contacts",
                    "className": "nav-item",
                    "replace": false,
                    "to": "/contacts",
                  },
                  "ref": null,
                  "rendered": "Contacts",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="sign-out-link"
                    replace={false}
                    to="/"
                  >
                    Sign Out
                  </Link>,
                  "className": "sign-out",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign Out",
                    "className": "sign-out-link",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Sign Out",
                  "type": [Function],
                },
                "type": "button",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="google-link"
            href="https://weather.com/"
          >
            EXIT
          </a>,
          "className": "exit",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "EXIT",
            "className": "google-link",
            "href": "https://weather.com/",
          },
          "ref": null,
          "rendered": "EXIT",
          "type": "a",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <nav
            className="nav-bar"
          >
            <h1
              className="cover-title"
            >
              <Link
                className="nav-bar-brand-link"
                replace={false}
                to="/main"
              >
                Cover
              </Link>
            </h1>
            <ul
              className="nav-menu"
            >
              <button
                className="hamburger"
                onClick={[Function]}
              />
              <li
                className="nav-item-li"
              >
                <Link
                  className="nav-item"
                  replace={false}
                  to="/main"
                >
                  Home
                </Link>
              </li>
              <li
                className="nav-item-li"
              >
                <Link
                  className="nav-item"
                  replace={false}
                  to="/contacts"
                >
                  Contacts
                </Link>
              </li>
              <button
                className="sign-out"
                onClick={[Function]}
              >
                <Link
                  className="sign-out-link"
                  replace={false}
                  to="/"
                >
                  Sign Out
                </Link>
              </button>
            </ul>
          </nav>,
          <button
            className="exit"
          >
            <a
              className="google-link"
              href="https://weather.com/"
            >
              EXIT
            </a>
          </button>,
        ],
        "className": "app-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="cover-title"
              >
                <Link
                  className="nav-bar-brand-link"
                  replace={false}
                  to="/main"
                >
                  Cover
                </Link>
              </h1>,
              <ul
                className="nav-menu"
              >
                <button
                  className="hamburger"
                  onClick={[Function]}
                />
                <li
                  className="nav-item-li"
                >
                  <Link
                    className="nav-item"
                    replace={false}
                    to="/main"
                  >
                    Home
                  </Link>
                </li>
                <li
                  className="nav-item-li"
                >
                  <Link
                    className="nav-item"
                    replace={false}
                    to="/contacts"
                  >
                    Contacts
                  </Link>
                </li>
                <button
                  className="sign-out"
                  onClick={[Function]}
                >
                  <Link
                    className="sign-out-link"
                    replace={false}
                    to="/"
                  >
                    Sign Out
                  </Link>
                </button>
              </ul>,
            ],
            "className": "nav-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="nav-bar-brand-link"
                  replace={false}
                  to="/main"
                >
                  Cover
                </Link>,
                "className": "cover-title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "Cover",
                  "className": "nav-bar-brand-link",
                  "replace": false,
                  "to": "/main",
                },
                "ref": null,
                "rendered": "Cover",
                "type": [Function],
              },
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="hamburger"
                    onClick={[Function]}
                  />,
                  <li
                    className="nav-item-li"
                  >
                    <Link
                      className="nav-item"
                      replace={false}
                      to="/main"
                    >
                      Home
                    </Link>
                  </li>,
                  <li
                    className="nav-item-li"
                  >
                    <Link
                      className="nav-item"
                      replace={false}
                      to="/contacts"
                    >
                      Contacts
                    </Link>
                  </li>,
                  <button
                    className="sign-out"
                    onClick={[Function]}
                  >
                    <Link
                      className="sign-out-link"
                      replace={false}
                      to="/"
                    >
                      Sign Out
                    </Link>
                  </button>,
                ],
                "className": "nav-menu",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "className": "hamburger",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-item"
                      replace={false}
                      to="/main"
                    >
                      Home
                    </Link>,
                    "className": "nav-item-li",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Home",
                      "className": "nav-item",
                      "replace": false,
                      "to": "/main",
                    },
                    "ref": null,
                    "rendered": "Home",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-item"
                      replace={false}
                      to="/contacts"
                    >
                      Contacts
                    </Link>,
                    "className": "nav-item-li",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Contacts",
                      "className": "nav-item",
                      "replace": false,
                      "to": "/contacts",
                    },
                    "ref": null,
                    "rendered": "Contacts",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="sign-out-link"
                      replace={false}
                      to="/"
                    >
                      Sign Out
                    </Link>,
                    "className": "sign-out",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign Out",
                      "className": "sign-out-link",
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "Sign Out",
                    "type": [Function],
                  },
                  "type": "button",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="google-link"
              href="https://weather.com/"
            >
              EXIT
            </a>,
            "className": "exit",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "EXIT",
              "className": "google-link",
              "href": "https://weather.com/",
            },
            "ref": null,
            "rendered": "EXIT",
            "type": "a",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
